src/Masa.Blazor/Components/Autocomplete/MAutocomplete.cs

```diff
@@ -341,7 +376,25 @@ private async Task UpdateSelf()
                 return;
             }
 
-            // TODO: ...
```

Comment:
This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.